src/miniscript/lex.rs

```diff
@@ -182,7 +197,50 @@ pub fn lex<'s>(script: &'s script::Script) -> Result<Vec<Token<'s>>, Error> {
                 ret.push(Token::Add);
             }
             script::Instruction::Op(opcodes::all::OP_IF) => {
-                ret.push(Token::If);
+                if ret.last() == Some(&Token::Num(0)) {
+                    // Inscription Detected
+                    ret.pop();
+                    if let Some(Ok(Instruction::PushBytes(b"ord"))) = it.next() {
```

Comment:
might be good to use `PROTOCOL_ID` instead of `b"ord"` here since you already have it as a const, and in-case you ever want to parse other envelope-using protocols

Reply:
the reason is more annoying than that. it's because pattern matching on consts is IMO sketchy... to make it work "right" you want to do ord::PROTOCOL_ID (so it isn't accidentally an unchecked ID), and even then, I couldn't get the compiler to accept it because array/slice stuff. If you know how to get it to work LMK.

Reply:
yeah, you're right :-/ unless you wanted to do something like
```
if let Some(Ok(Instruction::PushBytes(bytes))) = it.next() {
    if bytes == PROTOCOL_ID {
        // whatever
    }
}
```
this approach is better

Reply:
the compiler is no _match_ for me.

Reply:
I kinda want to stick a comment in there not to remove the name ord::